models.py

```diff
@@ -108,4 +108,74 @@ def forward(self, x):
 
         out = self.conv_last(x)
 
-        return out
\ No newline at end of file
+        return out
+
+class ResNetUNet(nn.Module):
+  def __init__(self, n_class):
+    super().__init__()
+
+    self.base_model = torchvision.models.resnet18(pretrained=True)
+    self.base_layers = list(self.base_model.children())
+
+    self.layer0 = nn.Sequential(*self.base_layers[:3]) # size=(N, 64, x.H/2, x.W/2)
+    self.layer0_1x1 = conv_relu(64, 64, 1, 0)
+    self.layer1 = nn.Sequential(*self.base_layers[3:5]) # size=(N, 64, x.H/4, x.W/4)
+    self.layer1_1x1 = conv_relu(64, 64, 1, 0)
+    self.layer2 = self.base_layers[5]  # size=(N, 128, x.H/8, x.W/8)
+    self.layer2_1x1 = conv_relu(128, 128, 1, 0)
+    self.layer3 = self.base_layers[6]  # size=(N, 256, x.H/16, x.W/16)
+    self.layer3_1x1 = conv_relu(256, 256, 1, 0)
+    self.layer4 = self.base_layers[7]  # size=(N, 512, x.H/32, x.W/32)
+    self.layer4_1x1 = conv_relu(512, 512, 1, 0)
+
+    self.upsample = nn.Upsample(scale_factor=2, mode='bilinear', align_corners=True)
+
+    self.conv_up3 = conv_relu(256 + 512, 512, 3, 1)
+    self.conv_up2 = conv_relu(128 + 512, 256, 3, 1)
+    self.conv_up1 = conv_relu(64 + 256, 256, 3, 1)
+    self.conv_up0 = conv_relu(64 + 256, 128, 3, 1)
+
+    self.conv_original_size0 = conv_relu(3, 64, 3, 1)
```

Comment:
Les images médicales ont 1 seul channel.
```suggestion
    self.conv_original_size0 = conv_relu(1, 64, 3, 1)
```